app/src/main/java/com/codepath/apps/critterfinder/activities/PetSearchFilterActivity.java

```diff
@@ -269,6 +306,30 @@ public void onClick(MaterialDialog dialog, DialogAction which) {
                 show();
     }
 
+    private void changeBreed(final SearchCriteria searchCriteria, final int position) {
+        new MaterialDialog.Builder(this)
+                .title(searchCriteria.title)
+                .items(mBreeds.get(mSearchFilter.getSpecies()))
+                .itemsCallbackMultiChoice(null, new MaterialDialog.ListCallbackMultiChoice() {
+                    @Override
+                    public boolean onSelection(MaterialDialog dialog, Integer[] which, CharSequence[] text) {
+                        if (which == null || which.length == 0) {
+                            mSearchFilter.setBreed(null);
+                            searchCriteria.value = "";
+                        } else {
+                            Breed newBreed = mBreeds.get(mSearchFilter.getSpecies()).get(which[0]);
+                            searchCriteria.value = newBreed.getName();
+                            mSearchFilter.setBreed(newBreed);
+                            mSearchCriteriaAdapter.notifyItemChanged(position);
+                        }
+                        mSearchCriteriaAdapter.notifyItemChanged(position);
+                        return false;
+                    }
+                }).
+                positiveText(getString(R.string.filter_choose)).
+                show();
```

Comment:
the material design dialog framework we're using lets us pass in an array of objects that have a toString method on them (which our Breed model does!) and it turns them into a nice list view.